### PR TITLE
[WOR-1800] Convert DeleteConfirmationModal and billing project usage to TSX

### PR DESCRIPTION
**Pull request by cahrens**
### Jira Ticket: https://broadworkbench.atlassian.net/browse/WOR-1800

Converts `DeleteConfirmationModal` from JS to TSX, and adds test coverage to it. We probably want to promote this file to our shared package (and write a Story for it), but I decided to leave that for a future PR.

Converts `DeleteBillingProjectModal` to TSX. Did not write a test for it, as it's basically just leveraging the `DeleteConfirmationModal`.

No changes in behavior intended.

Before:
image

After:
image

### CODE REVIEW

File: src/components/common/DeleteConfirmationModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        confirmationPrompt='confirm'
+        objectType='nut'
+        objectName='pistachio'
+        onConfirm={jest.fn()}
+        onDismiss={jest.fn()}
+      />
+    );
+    expect(isConfirmButtonDisabled()).toBe('true');
+    const confirmationInput = screen.getByLabelText('Type "confirm" to continue:');
+    await user.type(confirmationInput, 'conFIRM');
```
Comment (cahrens) [inline src/components/common/DeleteConfirmationModal.test.tsx:77]: I don't think there are any uses of this in the product (the "Delete Workspace" modal, which requires confirmation, is not using this shared modal). However, I hacked the code to test it out locally:

image

Comment (aherbst-broad) [reply src/components/common/DeleteConfirmationModal.test.tsx:77]: This may be the modal a user gets when deleting a group.

**Review by aherbst-broad — APPROVED** [review]
Very cool, nice to get tests around all this too.

File: src/components/common/DeleteConfirmationModal.test.tsx (hunk 1 above)
Comment (cahrens) [reply src/components/common/DeleteConfirmationModal.test.tsx:77]: It is, but it doesn't force the confirmation input field (doesn't pass confirmationPrompt).

**Review by trholdridge — APPROVED** [review]

**Pull request merged**